src/TraceEvent/TraceLoggingEventId.cs

```diff
@@ -38,8 +40,8 @@ public void Dispose()
             if (m_traceLoggingEventMap == null)
                 return;
 
-            foreach (var key in m_traceLoggingEventMap.Keys)
-                Marshal.FreeHGlobal((IntPtr)key.Provider);
+            foreach (var kvp in m_traceLoggingEventMap)
+                Marshal.FreeHGlobal((IntPtr)kvp.Key.Provider);
```

Comment:
@sharwell Pedantic: You can't deconstruct a KeyValuePair like that. You'll need a custom Deconstruct method.